docs/data/charts/zoom-and-pan/ZoomOverview.tsx

```diff
@@ -0,0 +1,49 @@
+import { ScatterChartPro } from '@mui/x-charts-pro/ScatterChartPro';
+import * as React from 'react';
+
+const dataLength = 800;
+const data = Array.from({ length: dataLength }).map((_, i) => ({
+  x: i,
+  y: 50 + Math.sin(i / 5) * 25,
+}));
+const series2Data = Array.from({ length: dataLength }).map((_, i) => ({
+  x: i,
+  y: 50 + Math.sin(i / 5) * 25,
+}));
+
+const xData = data.map((d) => d.x);
+
+export default function ZoomOverview() {
+  return (
+    <ScatterChartPro
+      xAxis={[
+        {
+          id: 'x',
+          data: xData,
+          zoom: {
+            filterMode: 'discard',
+            minSpan: 10,
+            panning: true,
+            overview: { enabled: true },
```

Comment:
Should we provide it as a slot instead?

Reply:
Yes

Reply:
I agree it could improve tree-shaking. But I'm not sure about the gain from a DX point of view.

To get the feature, users will have to pass it as a slot **and** enable it in the axes config.

The modification of the axes config is required to accurately compute the extra space needed to plot the zoom overview

If it's the case, I'm not in favor of a slot, because

- A user who wants advanced performances should create their own charts with composition to remove what is useless
- A user who just want something that works should use the charts out of the box

Reply:
Yeah, those are good point. We would also be coupling the chart size with the existence of slots, which I'm not sure would work well. 

I'll keep the current API. This is marked as unstable, so we can still change it if needed.